features/bsm/service/impl/src/main/java/org/opennms/netmgt/bsm/service/internal/DefaultBusinessServiceStateMachine.java

```diff
@@ -356,4 +364,14 @@ public String transform(GraphEdge edge) {
             m_rwLock.readLock().unlock();
         }
     }
+
+    @Override
+    public BusinessServiceGraph getGraph() {
+        m_rwLock.readLock().lock();
+        try {
+            return m_g;
```

Comment:
As discuessed, at the moment there is no easy/nice way to create a copy of the graph. 
I added a java doc comment to inform potential callers about not modifiing the graph. 
We need to re-visit this at same point in the future.